client/hwlib/src/collectors/cpuid.rs

```diff
@@ -0,0 +1,143 @@
+/* Copyright 2024 Canonical Ltd.
+ * All rights reserved.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+use anyhow::{anyhow, Result};
+use smbioslib;
+use std::fmt::{self, Display};
+
+#[derive(Clone, Debug)]
+pub struct CpuId([u8; 8]);
+
+impl CpuId {
+    pub fn from_smbios(proc_info: &smbioslib::SMBiosProcessorInformation) -> Result<Self> {
+        let cpu_id = proc_info
+            .processor_id()
+            .ok_or_else(|| anyhow!("processor ID not found"))?;
+        Ok(Self(*cpu_id))
+    }
+
+    pub fn codename(&self) -> Option<String> {
+        if self.0.is_empty() {
+            return None;
+        }
+        let cpu_id_hex = format!("0x{:x}{:02x}{:02x}", self.0[2], self.0[1], self.0[0]);
```

Comment:
We use only three bytes because the remaining once are used for other flags, not for a codename indication

Reply:
I've tried the proposed approach, and it takes too much effort to make it work, because we check `if ids.contains(&trimmed_cpu_id) ` as defined in the original logic, so we have to do conversion to string anyway. I guess it would be easier to keep the current approach